src/tools/runk/libcontainer/src/container.rs

```diff
@@ -58,18 +99,45 @@ impl Container {
                 kill(pid, signal)?;
             }
         } else {
-            if self.state == ContainerState::Stopped {
-                return Err(anyhow!("container {} not running", self.status.id));
-            }
             let pid = Pid::from_raw(self.status.pid);
             if status::is_process_running(pid)? {
                 kill(pid, signal)?;
             }
         }
+        // For cgroup v1, killing a process in a frozen cgroup does nothing until it's thawed. Only thaw the cgroup for SIGKILL.
```

Comment:
Good catch!

Nit: This comment is a bit long, so it would be nice to start a new line as necessary.